osu.Game.Tests/Visual/UserInterface/TestSceneChatOverlayScenarios.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Testing;
+using osu.Game.Online.Chat;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Chat.Selection;
+using osu.Game.Overlays.Chat.Tabs;
+using osu.Game.Users;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Tests.Visual.UserInterface
+{
+    public class TestSceneChatOverlayScenarios : ManualInputManagerTestScene
+    {
+        public override IReadOnlyList<Type> RequiredTypes => new[]
+        {
+            typeof(ChannelTabControl),
+            typeof(ChannelTabItem),
+            typeof(ChatOverlay),
+        };
+
+        private TestChatOverlay chatOverlay;
+
+        [Cached]
+        private ChannelManager channelManager = new ChannelManager();
+
+        private Channel channel1;
+        private Channel channel2;
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            var availableChannels = (BindableList<Channel>)channelManager.AvailableChannels;
+
+            availableChannels.Add(channel1 = new Channel(new User()) { Name = "test1" });
+            availableChannels.Add(channel2 = new Channel(new User()) { Name = "test2" });
+
+            Add(chatOverlay = new TestChatOverlay
+            {
+                RelativeSizeAxes = Axes.Both,
+                Size = new Vector2(1)
+            });
+        }
+
+        [SetUpSteps]
+        public void SetUpSteps()
```

Comment:
Are you doing this to start from a fresh slate? If so, a better way is:

Add `[Cached]` to `ChannelManager`'s class then use

```csharp
[SetUp]
public void SetUp() => Schedule(() =>
{
	Child = new Container
    {
		channelManager = new ChannelManager()
        {
            AvailableChannels = { Values = { new []
				{
					...
 				}
			}
        }
		chatOverlay = new TestChatOverlay
        {
            RelativeSizeAxes = Axes.Both,
            Size = new Vector2(1)
        }
    }
});
```

and remove all `SetUpSteps`.